openpype/hosts/max/api/preview_animation.py

```diff
@@ -0,0 +1,309 @@
+        rt.preferences.playPreviewWhenDone = False
+        yield
+    finally:
+        rt.viewport.setCamera(original)
+        rt.preferences.playPreviewWhenDone = has_autoplay
+
+
+@contextlib.contextmanager
+def viewport_preference_setting(general_viewport,
+                                nitrous_viewport,
+                                vp_button_mgr):
+    """Function to set viewport setting during context
+    ***For Max Version < 2024
+    Args:
+        camera (str): Viewport camera for review render
+        general_viewport (dict): General viewport setting
+        nitrous_viewport (dict): Nitrous setting for
+            preview animation
+        vp_button_mgr (dict): Viewport button manager Setting
+        preview_preferences (dict): Preview Preferences Setting
+    """
+    orig_vp_grid = rt.viewport.getGridVisibility(1)
+    orig_vp_bkg = rt.viewport.IsSolidBackgroundColorMode()
+
+    nitrousGraphicMgr = rt.NitrousGraphicsManager
+    viewport_setting = nitrousGraphicMgr.GetActiveViewportSetting()
+    vp_button_mgr_original = {
+        key: getattr(rt.ViewportButtonMgr, key) for key in vp_button_mgr
+    }
+    nitrous_viewport_original = {
+        key: getattr(viewport_setting, key) for key in nitrous_viewport
+    }
+
+    try:
+        rt.viewport.setGridVisibility(1, general_viewport["dspGrid"])
+        rt.viewport.EnableSolidBackgroundColorMode(general_viewport["dspBkg"])
+        for key, value in vp_button_mgr.items():
+            setattr(rt.ViewportButtonMgr, key, value)
+        for key, value in nitrous_viewport.items():
+            if nitrous_viewport[key] != nitrous_viewport_original[key]:
+                setattr(viewport_setting, key, value)
+        yield
+
+    finally:
+        rt.viewport.setGridVisibility(1, orig_vp_grid)
+        rt.viewport.EnableSolidBackgroundColorMode(orig_vp_bkg)
+        for key, value in vp_button_mgr_original.items():
+            setattr(rt.ViewportButtonMgr, key, value)
+        for key, value in nitrous_viewport_original.items():
+            setattr(viewport_setting, key, value)
+
+
+def publish_review_animation(instance, staging_dir, start,
+                             end, ext, fps, viewport_options):
+    """Function to set up preview arguments in MaxScript.
+    ****For 3dsMax 2024+
+
+    Args:
+        instance (str): instance
+        filepath (str): output of the preview animation
+        start (int): startFrame
+        end (int): endFrame
+        fps (float): fps value
+        viewport_options (dict): viewport setting options
+
+    Returns:
+        list: job arguments
+    """
+    job_args = list()
+    filename = "{0}..{1}".format(instance.name, ext)
+    filepath = os.path.join(staging_dir, filename)
+    filepath = filepath.replace("\\", "/")
+    default_option = f'CreatePreview filename:"{filepath}"'
+    job_args.append(default_option)
+    frame_option = f"outputAVI:false start:{start} end:{end} fps:{fps}" # noqa
```

Comment:
Does `fps` actually mean anything in this context if `outputAVI:false` - I assume we're outputting images? Are we sure `fps` means anything here?

Reply:
It is supposed not to be there. But I would keep the fps data inside the collector as this needs to pass through extract review from the OP pipeline.